### ISSUE TITLE
BBC: Blead Breaks Locale::Maketext::Utils

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#20454]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.43 running under perl 5.37.6.

-----------------------------------------------------------------

BBC: Blead Breaks Locale::Maketext::Utils

Please see http://matrix.cpantesters.org/?dist=Locale::Maketext::Utils

---
**Flags**
- category=core
- severity=low
---
**Perl configuration**
```
Site configuration information for perl 5.37.6:

Configured by cpan at Sat Oct 29 01:39:32 EDT 2022.

Summary of my perl5 (revision 5 version 37 subversion 6) configuration:
  Commit id: b5132f3c1c423921c2b76c1ac425475f14e5d4fa
  Platform:
    osname=linux
    osvers=5.15.75-0-lts
    archname=x86_64-linux
    uname='linux cjg-alpine3 5.15.75-0-lts #1-alpine smp thu, 27 oct 2022 14:50:14 +0000 x86_64 linux '
    config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2'
    cppflags='-D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong'
    ccversion=''
    gccversion='11.2.1 20220219'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib /lib
    libs=-lpthread -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=/usr/lib/libc.a
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


---
@INC for perl 5.37.6:
    /home/cpan/bin/perl-blead/lib/site_perl/5.37.6/x86_64-linux
    /home/cpan/bin/perl-blead/lib/site_perl/5.37.6
    /home/cpan/bin/perl-blead/lib/5.37.6/x86_64-linux
    /home/cpan/bin/perl-blead/lib/5.37.6

---
Environment for perl 5.37.6:
    HOME=/home/cpan
    LANG=C.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl-blead/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash
```

**Comment by jkeenan** [comment]
> This is a bug report for perl from "Carlos Guevara" [carlos@carlosguevara.com](mailto:carlos@carlosguevara.com), generated with the help of perlbug 1.43 running under perl 5.37.6.
> 
> BBC: Blead Breaks Locale::Maketext::Utils
> 
> Please see http://matrix.cpantesters.org/?dist=Locale::Maketext::Utils

I have been unable to use `Porting/bisect.pl` to identify the first bad commit that caused this test failure.  I've tried various invocations like this:
```
perl Porting/bisect.pl \
--module=Locale::Maketext::Utils \
--start=v5.37.4 \
--end=382a56637961ef97fde5808e97df108db4e3ed8d
```
... but no matter how far back I go I get a result like this:
```
make: stopped in /tmp/a4Co5F6G70/Locale-Maketext-Utils-0.42-0
  DMUEY/Locale-Maketext-Utils-0.42.tar.gz
  /usr/bin/make test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports DMUEY/Locale-Maketext-Utils-0.42.tar.gz
Died at -e line 2.
HEAD is now at f2582f5b18 add perlhist entry
bad - non-zero exit from /tmp/dTfc1z9XtG/bin/perl -I /home/jkeenan/.cpan -MCPAN::MyConfig -MCPAN -e $CPAN::Config->{build_dir}=q{/tmp/a4Co5F6G70}; -e install('Locale::Maketext::Utils'); die unless CPAN::Shell->expand(Module => 'Locale::Maketext::Utils')->uptodate;
Runner returned 256, not 0 for start revision at Porting/bisect.pl line 240.
```
This distribution has a ton of prerequisites and simply to get to this point (trying the `end` and `start` revisions) takes 32 minutes.  If anyone wants to take a crack at this and install the prerequisites, I have a stripped-down version of the failing test program that may make debugging easier.  But, other than guessing that this is a locale-related problem, I don't know where we go next.

**Comment by jkeenan** [comment]
Blead did not break Locale::Maketext::Utils.  Changes in DateTime* broke Locale::Maketext::Utils.

As noted in earlier posts, Locale-Maketext-Utils has a large number of prerequisites, among them DateTime-Locale, DateTime-TimeZone and DateTime.  New versions of each were released between Oct 23 and Oct 28.  The [Changes file for DateTime-Locale](https://metacpan.org/dist/DateTime-Locale/changes) says that in version 1.37, released Oct 23, "Rebuilt all locale data with the data from CLDR 42.0.0."  The CPANtesters failures began appearing on Oct 23 -- and any CPANtesters report run after that date -- regardless of the `perl` release it's run with -- will pick up these most recent DateTime* versions and trigger failures in Locale-Maketext-Utils.

I have filed a [ticket](https://rt.cpan.org/Ticket/Display.html?id=144987) in the bug tracker for Locale-Maketext-Utils.  @drmuey
